src/raftstore/store/peer.rs

```diff
@@ -802,6 +813,83 @@ impl Peer {
         cmd.call(resp);
     }
 
+    fn is_uptodate(&self, matched_index: u64) -> bool {
```

Comment:
Why abstract this as a function? And I don't think when `matched_index >= truncated_index` means the progress is update to date.